test/test-trace-api.js

```diff
@@ -180,7 +181,13 @@ describe('Trace Interface', function() {
       traceAPI.runInRootSpan({name: 'root', url: 'root'}, function(rootSpan) {
         var id = traceAPI.getCurrentContextId();
         assert.strictEqual(id, rootSpan.trace.traceId);
-      });      
+      });
+    });
+
+    it('should return get the project ID if set in config', function() {
```

Comment:
It would be good to have a test that checks that project id becomes available after asynchronous resolution (currently this behavior relies on the trace api and trace writer sharing the same mutable config object which may not always be the case). I don't want to hold up this PR though, maybe you could add a TODO?

Reply:
Yeah, that's a good point. Just added a test case to `test-index.js` as `index.js` is where the shared `config` assignment takes place.

Reply:
Hmm, the test looks good but it is failing. I can try to help debug next week.

Reply:
Turned out I needed to clear the project environment variable.